mars-rover-new/src/mars_rover_new/core.clj

```diff
@@ -1,7 +1,94 @@
+;; Mars Rover Exercise
+
 (ns mars-rover-new.core
+  (:require [clojure.string :as string])
   (:gen-class))
 
+;; todo: input validation, check invalid moves
+
+;; input validation: check for number of args, catch errors during type conversion
+
+(declare add-plateau-size, add-rover-position, add-cmd-sequence)
```

Comment:
You won't have to `declare` these non-binded vars if you move the `user-input` function at the bottom. 

In my experience, the function which ties a lot of functions together are written at the bottom so that all the downstream functions are declared before that larger function is called.

Reply:
Makes sense, done!